### PR TITLE
Unathi Guns & Armor

### PR DESCRIPTION
**Pull request by RustingWithYou**
Adds several pre-contact Unathi gun designs, some of which currently accessible through antag gear crates and the Unathi pirate ghost ships. All gun sprites done by Gecko.

Additionally buffs the generic Unathi armor and Hegemony armor to be more on par with standard plate carriers instead of just being good at melee and terrible at everything else, as well as respriting the generic armor and rewriting flavor text. Armor resprites done by Dronz.

### CODE REVIEW

**Comment by RustingWithYou** [comment]
!review

File: code/modules/projectiles/ammunition/bullets.dm [hunk 1]
```diff
@@ -223,6 +223,14 @@
 	spent_icon = "trackingshell-spent"
 	projectile_type = /obj/item/projectile/bullet/tracking
 
+/obj/item/ammo_casing/shotgun/moghes
+	name = "wall shell"
+	desc = "The appropriately-named wall shell, made specifically for the Moghesian wall gun."
+	desc_extended = "This complicated and expensive shotfun shell fires both a solid slug and a hail of shots at the same time, making it a destructive and dangerous ammunition at all ranges, assuming the user can point in the target's general direction. Somehow it seems to somewhat fit in other shotgun type weaponry too."
```
Comment (dessysalta) [inline code/modules/projectiles/ammunition/bullets.dm:229]: Misspell, shot**fun** instead of shot**gun**

Comment (RustingWithYou) [reply code/modules/projectiles/ammunition/bullets.dm]: i mean, it is fun too probably.

**Review by NonQueueingMatt — APPROVED** [review]

File: code/modules/projectiles/guns/projectile/rifle.dm [hunk 2]
```diff
@@ -418,3 +418,25 @@
...
+	allowed_magazines = list(/obj/item/ammo_magazine/crackrifle)
+	load_method = MAGAZINE
+
+/obj/item/gun/projectile/shotgun/pump/rifle/unathi/handle_pump_loading()
+	if(ammo_magazine && length(ammo_magazine.stored_ammo))
+		var/obj/item/ammo_casing/AC = ammo_magazine.stored_ammo[1] //load next casing.
+		if(AC)
+			AC.forceMove(src)
+			ammo_magazine.stored_ammo -= AC
+			chambered = AC
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/rifle.dm:442]: This looks like to be more adapt to be a subtype of shotguns, ie pump/rifle/magazine_fed? Also, what type is `/obj/item/gun/projectile/shotgun/pump/rifle/unathi`? I do not see it in the code, the one above is called crackrifle?

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 3]
```diff
@@ -278,3 +278,70 @@
...
+		open = FALSE
+		update_icon()
+
+/obj/item/gun/projectile/shotgun/wallgun/unload_ammo(user, allow_dump)
+	if(open)
+		..(user, allow_dump=1)
+
+/obj/item/gun/projectile/shotgun/wallgun/load_ammo(obj/item/A, mob/user)
+	if(!open)
+		to_chat(user, "<span class='warning'>You need to open the cover to load [src].</span>")
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/shotgun.dm:324]: SPAN_WARNING

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 4]
```diff
@@ -278,3 +278,70 @@
...
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(H.mob_size <10)
+				H.visible_message(SPAN_WARNING("\The [src] flies out of \the [H]'s' hand!"), SPAN_WARNING("\The [src] flies out of your hand!"))
+				H.drop_item(src)
+				src.throw_at(get_edge_target_turf(src, GLOB.reverse_dir[H.dir]), 4, 4)
+
+				var/obj/item/organ/external/LH = H.get_organ(BP_L_HAND)
+				var/obj/item/organ/external/RH = H.get_organ(BP_R_HAND)
+				var/active_hand = H.hand
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/shotgun.dm:342]: Useless assignment, you only access it once

File: code/modules/projectiles/projectile.dm [hunk 5]
```diff
@@ -101,6 +101,7 @@
 	var/anti_materiel_potential = 1 //how much the damage of this bullet is increased against mechs
 
 	var/iff // identify friend or foe. will check mob's IDs to see if they match, if they do, won't hit
+	var/secondary_projectile // does this projectile actually launch two projectiles?
```
Comment (FluffyGhoster) [inline code/modules/projectiles/projectile.dm:104]: Properly DMDoc this

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 6]
```diff
@@ -278,3 +278,70 @@
...
+	ammo_type = /obj/item/ammo_casing/shotgun/moghes
+	load_method = SINGLE_CASING|SPEEDLOADER
+	w_class = ITEMSIZE_NORMAL
+	fire_delay = ROF_INTERMEDIATE
+	force = 5
+	max_shells = 1
+	caliber = "shotgun"
+	fire_sound = 'sound/weapons/gunshot/gunshot_shotgun2.ogg'
+	handle_casings = HOLD_CASINGS
+	var/open = FALSE
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/shotgun.dm:301]: DMDoc this

File: code/modules/projectiles/guns/projectile/pistol.dm [hunk 7]
```diff
@@ -553,3 +553,25 @@
...
+	icon_state = "spitterpistol"
+	item_state = "spitterpistol"
+	caliber = "11.6mm"
+	fire_sound = 'sound/weapons/gunshot/gunshot_pistol.ogg'
+	load_method = MAGAZINE
+	fire_delay = ROF_PISTOL
+
+/obj/item/gun/projectile/pistol/spitter/update_icon()
+	..()
+	if(ammo_magazine?.stored_ammo.len)
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/pistol.dm:574]: `length()` over .len

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/projectile/rifle.dm (hunk 2 above)
Comment (RustingWithYou) [reply code/modules/projectiles/guns/projectile/rifle.dm:442]: That was the original name before I refactored to be more clear, missed it in this proc.

Comment (RustingWithYou) [reply code/modules/projectiles/guns/projectile/rifle.dm]: Refactored this and other relevant guns to pump/rifle/magazine_fed

**Review by FluffyGhoster — APPROVED** [review]
Looks good enough to me

**Pull request merged**